### PR TITLE
add Google style docstring to web.py

### PR DESCRIPTION
**Pull request by GrugLife**
# Description

_Add Google style docstrings to web.py per the following link:
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Please let me know of any changes to the current docstrings.

If all looks good, I will tackle another file.
_

 #532 (web.py) 

## Status
**READY** 

## Type of change

- Documentation (fix or adds documentation)

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation (if applicable)
- [NA] I have added tests that prove my fix is effective or that my feature works
- [NA] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by jacobtomlinson** [comment]
Thanks for this!

The linter is raising some issues with having blank lines after your docstrings. Could you please address this.

```
lint runtests: commands[2] | pydocstyle opsdroid tests
opsdroid/web.py:35 in public method `get_port`:
        D202: No blank lines allowed after function docstring (found 1)
opsdroid/web.py:56 in public method `get_host`:
        D202: No blank lines allowed after function docstring (found 1)
opsdroid/web.py:74 in public method `get_ssl_context`:
        D202: No blank lines allowed after function docstring (found 1)
opsdroid/web.py:115 in public method `build_response`:
        D202: No blank lines allowed after function docstring (found 1)
opsdroid/web.py:152 in public method `web_index_handler`:
        D202: No blank lines allowed after function docstring (found 1)
```

**Comment by FabioRosado** [comment]
I am a bit confused by the linting issues, with the google style docstrings you should have the docstrings like:

```python
    def build_url(self, method):
        """Build the url to connect with api.

        Helper function to build the url to interact with the
        Rocket.Chat REST API. Uses the global variable API_PATH
        that points to current api version. (example: /api/v1/)

        Args:
            method (string): Api call endpoint.

        Return:
            String that represents full API url.

        """
```

Unless pylint got updated and they don't accept this style anymore?

**Comment by jacobtomlinson** [comment]
But there shouldn't be a blank line between the last `"""` and the first line of code.

**Comment by FabioRosado** [comment]
Oh you are right haha I thought that the lint was complaining about the line between the first docstring line and the Args, bit.

**Comment by GrugLife** [comment]
Sure thing, will get on it.

File: opsdroid/web.py [hunk 1]
```diff
@@ -113,12 +145,28 @@ def setup_webhooks(self, skills):
...
 
     async def web_stats_handler(self, request):
-        """Handle stats request."""
+        """Handle stats request.
+
+        Args:
+            request: web request to render opsdroid stats
+
+        Returns:
+            dict: returns successful status code and dictionary with stats requested
```
Comment (jacobtomlinson) [inline opsdroid/web.py]: Also looks like this line is too long. Could you add a linebreak to make it less that 80 characters.

**Comment by GrugLife** [comment]
agggh, I thought I had it.

Is there a way to check for this before I push?  I ran it through pylint and it checked out.

Thanks,
Greg

**Comment by FabioRosado** [comment]
@GrugLife No worries, its all a learning experience 😄 👍 
Yeah you can check this if you run tox from your command line (you might need to pip install it).

Basically `cd` into your opsdroid directory, type `tox` and the tests will run, you will see all the tests that passed/failed, the % of test coverage and which lines are not covered and any linting issues you might have

**Comment by jacobtomlinson** [comment]
🎉 Thanks for bearing with us! We have quite strict linting rules in this project but I'm glad to see we got there in the end.

Ping me your address via DM on [Twitter](https://twitter.com/_jacobtomlinson) or [Gitter](https://gitter.im/opsdroid) and I'll post you some [opsdroid stickers](https://medium.com/opsdroid/stickers-for-contributors-a0a1f9c30ec1)!

**Comment by GrugLife** [comment]
No worries, it's a good learning experience to work within the linting rules.  And thanks for bearing with me!

If you don't mind, I would like to do the same on another file.

**Comment by jacobtomlinson** [comment]
Please go ahead!

**Pull request merged**